### DISCUSSION TITLE
extension hooks as a collection

### DISCUSSION THREAD

**Discussion by gensmann** [directus/directus#3468] [Feature Requests]
Extension hooks could benefit from a system collection just like web hooks, so it was possible to configure and create them through the GUI.

When saved/updated they could be persisted down into the file system as js, files. This would also make it possible to use the directus revision system for them and the role system for letting misc developers extend the system or specific collections only, or make suggestions without publicising them due to role limits.

A setting wether they should overwrite existing equally named physical files

This is of cause only feasible for one file extension hooks, but would be nice for small tweaks

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
